src/Lime.Protocol/LimeUri.cs

```diff
@@ -26,16 +27,25 @@ public LimeUri(string uriPath)
             if (Uri.IsWellFormedUriString(uriPath, UriKind.Absolute))
```

Comment:
Move the  relative URI check to the beginning of the method, since it is most common case in Lime protocol. And with that, you can join the new checks in line 36 in one place.

Reply:
I´m not sure I understood what you mean, but if I did, that would change the initial behavior of the method, which is something I'm not keen on doing.

Reply:
Something like this:

```suggestion
        if (!Uri.IsWellFormedUriString(uriPath, UriKind.Relative))
        {       
            // The URI may be absolute (including the protocol, host and path) or invalid.
            if (Uri.IsWellFormedUriString(uriPath, UriKind.Absolute))
            {
                _absoluteUri = new Uri(uriPath);
                ValidatLimeScheme(_absoluteUri);
            }
            else if (Uri.TryCreate(uriPath, UriKind.Absolute, out var receivedUri) &&
                ReceivedUriPathIsEncoded(uriPath) &&
                Uri.IsWellFormedUriString($"{receivedUri.Scheme}://{receivedUri.UserInfo}@{receivedUri.Host}", UriKind.Absolute) &&
                Uri.IsWellFormedUriString(receivedUri.PathAndQuery + receivedUri.Fragment, UriKind.Relative))
            {
                // TODO: This 'if' statement is only necessary while the related issue is not fixed
                // Issue: https://github.com/dotnet/runtime/issues/21626
                _absoluteUri = new Uri(uriPath);
                ValidatLimeScheme(_absoluteUri);
            }
            else
            {
                throw new ArgumentException("Invalid URI format");
            }
        }

        Path = uriPath.TrimEnd('/');
```

Reply:
I see your point and do agree that it makes for a cleaner code. That said, I am still not sure that we should change it, here's why:

- The intention of this PR is to fix a bug. I wrote the code with the intention of not changing previous functionality. I don't see a reason why your suggested change would introduce a different bug, but IT IS changing the original flow.
- In the best possible scenario, this code I wrote is going to be short-lived. We intend to remove it as soon as the dotnet issue is resolved. In that sense I believe your suggestion makes it harder for code removal (My code you just delete a couple of lines and that's it. In your suggested code we would delete those same lines and then would be left with a method that's just calling another method, which would then beg the question "WHY???" and then the developer would probably want to remove IsWellFormedAbsoluteUriString() method, sending line 47 back to the original code)

Reply:
Ok, no problem.